app/models/user.rb

```diff
@@ -9,4 +9,41 @@ class User < ApplicationRecord
   has_many :posts
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
+  has_many :friendships
+  has_many :inverse_friendships, class_name: 'Friendship', foreign_key: 'friend_id'
+
+  def friends
+    friends_array = friendships.map { |friendship| friendship.friend if friendship.confirmed }
+    friends_array.concat(inverse_friendships.map { |friendship| friendship.user if friendship.confirmed })
+    friends_array.compact
+  end
```

Comment:
The idea behind using double friendship records is to avoid doing multiple queries to generate the complete list of a user's friends. Instead of building a list by concatenating two smaller lists, you should do something like:

```
  def friends
    requested_friends
      .includes(:sent_friendships)
      .where('friendships.status = true')
  end
```

You may check out my own solution to this problem here, for inspiration: 
https://github.com/voscarmv/ror-social-scaffold/blob/master/app/models/user.rb